src/main/java/usecases/UserSettings.java

```
@@ -47,10 +47,42 @@ public UserList deserializeUsers() {
      * @param password password of new user
      * @param email email address of user
      * @param phoneNumber phone number of new user
-     * @return UserManager representing this new user
+     * @return UserManager representing this new user, null if failed to create account
      */
     public UserManager createAccount(String username, String password, String email, String phoneNumber) {
         UserManager newUser = new UserManager(username, password, email, phoneNumber);
+
+        // Verify the given info checks out
+        boolean error = false;
+        if (users.getUser(username) != null) {
+            // username is already taken
+            error = true;
+        }
+        if (!phoneNumber.matches("^\\(\\d\\d\\d\\)-\\d\\d\\d-\\d\\d\\d$")) {
+            // phone number fails regex of form: (416)-123-4567
+            error = true;
+        }
+        if (!email.matches("^\\w+@\\w+\\.((com)|(ca))$")) {
+            // email fails regex
+            error = true;
+        }
+        if (!users.phoneExists(phoneNumber)) {
+            // phone number already exists
+            error = true;
+        }
+        if (!users.emailExists(email)) {
+            // email already exists
+            error = true;
+        }
+
+        if (error) {
```

This should return an error message for the frontend to display.